src/Processors/ExpandEnums.php

```diff
@@ -73,20 +73,35 @@ private function expandSchemaEnum(Analysis $analysis): void
         ]);
 
         foreach ($schemas as $schema) {
-            if ($schema->enum !== Generator::UNDEFINED && is_string($schema->enum)) {
-                $source = $schema->enum;
-                // Convert to Enum value if it is an Enum class string
+            if ($schema->enum == Generator::UNDEFINED) {
+                continue;
+            }
+
+            if (is_string($schema->enum)) {
+                // might be enum class
                 if (is_a($schema->enum, '\UnitEnum', true)) {
-                    $enums = [];
-                    foreach ($source::cases() as $case) {
-                        $enums[] = $case->value ?? $case->name;
-                    }
+                    $source = $schema->enum::cases();
+                } else {
+                    throw new \InvalidArgumentException("Unexpected enum value, requires specifying the Enum class string: $schema->enum");
+                }
+            } elseif (is_array($schema->enum)) {
+                // might be array of enum, string, int, etc...
+                $source = $schema->enum;
+            } else {
+                // other type (do nothing)
```

Comment:
Should this also throw an exception? I would think this is an unsupported value?

At least some debug logging might be nice here.